client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchRoundDataEntry.tsx

```diff
@@ -0,0 +1,160 @@
+import { useDebounce } from '../../../utils/debounce'
+
+const Container = styled('div')`
+  display: flex;
+  flex-direction: column;
+  gap: 12px;
+  width: 100%;
+`
+
+interface IProps {
+  electionId: string
+  jurisdictionId: string
+  roundId: string
+}
+
+const BatchRoundDataEntry: React.FC<IProps> = ({
+  electionId,
+  jurisdictionId,
+  roundId,
+}) => {
+  const batchesQuery = useBatches(electionId, jurisdictionId, roundId)
+  const contestsQuery = useContestsJurisdictionAdmin(electionId, jurisdictionId)
+  const recordBatchResults = useRecordBatchResults(
+    electionId,
+    jurisdictionId,
+    roundId
+  )
+  const { confirm, confirmProps } = useConfirm()
+
+  const [searchQuery, setSearchQuery] = useState('')
+  const [debouncedSearchQuery] = useDebounce(searchQuery)
+  const [selectedBatchId, setSelectedBatchId] = useState<IBatch['id']>()
+  const [areChangesUnsaved, setAreChangesUnsaved] = useState(false)
+
+  const batches = batchesQuery.isSuccess ? batchesQuery.data.batches : []
+  const filteredBatches = batches.filter(batch =>
+    batch.name.toLowerCase().includes(debouncedSearchQuery.toLowerCase())
+  )
+  const selectedBatch = batches.find(batch => batch.id === selectedBatchId)
+
+  // Auto-select first batch on initial load
+  useEffect(() => {
+    if (!selectedBatchId && filteredBatches.length > 0) {
+      setSelectedBatchId(filteredBatches[0].id)
+    }
+  }, [filteredBatches, selectedBatchId, setSelectedBatchId])
+
+  // Auto-select first search match
+  useEffect(() => {
+    if (
+      debouncedSearchQuery &&
+      filteredBatches.length > 0 &&
+      !areChangesUnsaved
+    ) {
+      setSelectedBatchId(filteredBatches[0].id)
+    }
+  }, [
+    debouncedSearchQuery,
+    filteredBatches,
+    areChangesUnsaved,
+    setSelectedBatchId,
+  ])
+
+  if (!batchesQuery.isSuccess || !contestsQuery.isSuccess) {
+    return null
+  }
+
+  const areResultsFinalized = Boolean(batchesQuery.data.resultsFinalizedAt)
+  // Batch comparison audits only support a single contest
+  const [contest] = contestsQuery.data
+
+  const selectBatch = (batchId: string) => {
+    if (areChangesUnsaved) {
+      confirm({
+        title: 'Unsaved Changes',
+        description:
+          'You have unsaved changes. ' +
+          'Are you sure you want to leave this batch without saving changes?',
+        yesButtonLabel: 'Discard Changes',
+        yesButtonIntent: 'danger',
+        onYesClick: () => setSelectedBatchId(batchId),
+        noButtonLabel: 'Cancel',
+      })
+      return
+    }
+    setSelectedBatchId(batchId)
+  }
+
+  return (
+    <Container>
+      {areResultsFinalized && (
+        <Callout icon="tick-circle" intent="success">
+          Tallies finalized
+        </Callout>
+      )}
+      <ListAndDetail>
+        <List
+          search={{
+            placeholder: 'Search batches...',
+            setQuery: setSearchQuery,
+          }}
+        >
+          {filteredBatches.map(batch => (
```

Comment:
Could use an empty state when there are no matching batches. Maybe just a simple line of text: "No batches found"

Reply:
https://github.com/votingworks/arlo/pull/1661/commits/61c981128507343571a70dc6058c16554606a12c

<img width="265" alt="no-batches-found" src="https://user-images.githubusercontent.com/12616928/199332601-b2319deb-bbc0-414e-90b6-2f65f73086b8.png">